### PR TITLE
Add Bisheng JDK RISC-V support

### PR DESCRIPTION
**Pull request by feilongjiang**
Add Bisheng JDK RISC-V support for openjdk-build. Closes: #2420

### CODE REVIEW

**Review by M-Davies — COMMENTED** [review]
You're probably better off making any changes inside the `pipelines/` dir in https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/tree/master/pipelines as that dir is being [moved over there soon](https://github.com/AdoptOpenJDK/openjdk-build/pull/2463). I see the tests have passed and code looks good however so I will not block this if people want to merge this in before the code split

**Comment by feilongjiang** [comment]
Github actions didn't run risc-v build, should we run extra tests to verify my changes?

**Comment by feilongjiang** [comment]
> You're probably better off making any changes inside the `pipelines/` dir in https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/tree/master/pipelines as that dir is being [moved over there soon](https://github.com/AdoptOpenJDK/openjdk-build/pull/2463). I see the tests have passed and code looks good however so I will not block this if people want to merge this in before the code split

If this pr is merged after code splitting, I will remove changes under `pipelines`, thanks!

File: configureBuild.sh [hunk 1]
```diff
@@ -159,6 +159,8 @@ setRepository() {
     suffix="corretto/corretto-${BUILD_CONFIG[OPENJDK_CORE_VERSION]:3}"
   elif [[ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_DRAGONWELL}" ]]; then
     suffix="alibaba/dragonwell${BUILD_CONFIG[OPENJDK_CORE_VERSION]/jdk/}"
+  elif [[ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_BISHENG}" ]]; then
+    suffix="feilongjiang/bishengjdk-11-mirror"
```
Comment (aahlenst) [inline configureBuild.sh:163]: If I read this correctly, this code is pulled from https://github.com/feilongjiang/bishengjdk-11-mirror which in turn comes from https://gitee.com/openeuler/bishengjdk-11/tree/risc-v. So basically we run third-party code on our stateful machines. Has anyone verified that the necessary security practices are in place? Do we have guidelines for that?

(To be clear, this is not a criticism of the PR or its author. I know we've pulled code from elsewhere in the past.)

File: build-farm/platform-specific-configurations/linux.sh [hunk 2]
```diff
@@ -171,5 +170,15 @@ then
...
+elif [ "${ARCHITECTURE}" == "riscv64" && "${VARIANT}" == "${BUILD_VARIANT_BISHENG}" ]; then
+  echo Bisheng RISCV cross-compilation ... 
+  export RISCV64=/opt/riscv_toolchain_linux
+  export LD_LIBRARY_PATH=$RISCV64/lib64
+  export PATH="$RISCV64/bin:$PATH"
+  # riscv has to use a cross compiler
+  export CC=$RISCV64/bin/riscv64-unknown-linux-gnu-gcc
+  export CXX=$RISCV64/bin/riscv64-unknown-linux-gnu-g++
+  CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --openjdk-target=riscv64-unknown-linux-gnu --with-sysroot=/opt/fedora28_riscv_root --with-boot-jdk=$JDK_BOOT_DIR"
+  BUILD_ARGS="${BUILD_ARGS} -F"
```
Comment (karianna) [inline build-farm/platform-specific-configurations/linux.sh:182]: The whitespace in this block doesn't match the above, can you check tabs vs spaces and see which block needs changing?

File: configureBuild.sh (hunk 1 above)
Comment (karianna) [reply configureBuild.sh:163]: @aahlenst The security team will review the mirror.  There's some extra work to be done to also enable with build on Docker as std practice.

@feilongjiang Have you explored being able to build this in a Docker container?

**Review by karianna — CHANGES_REQUESTED** [review]
A couple of small comments but the rest of the code looks good to me.

**Review by M-Davies — APPROVED** [review]
Approved so long as Martijn's comments are addressed

File: build-farm/platform-specific-configurations/linux.sh (hunk 2 above)
Comment (feilongjiang) [reply build-farm/platform-specific-configurations/linux.sh:182]: The above code block uses tabs, I replaced it with spaces.

File: configureBuild.sh (hunk 1 above)
Comment (feilongjiang) [reply configureBuild.sh:163]: I only tested the native build on Ubuntu 18.04. Docker and cross-compile builds have not been tested yet.

**Review by karianna — APPROVED** [review]

File: configureBuild.sh (hunk 1 above)
Comment (karianna) [reply configureBuild.sh:163]: OK, can you see if you can get cross-compile and Docker builds going next?  We will eventually have a docker only build policy so might as well get ahead of that :-).

Comment (feilongjiang) [reply configureBuild.sh:163]: Ok, I'll try next

**Comment by tellison** [comment]
The AdoptOpenJDK build should switch to using the [OpenJDK sandbox RISC-V branch](https://github.com/openjdk/jdk-sandbox/tree/riscv-port-branch) when that is synced up with the Bisheng repository, as described by Yadong here:

https://mail.openjdk.java.net/pipermail/porters-dev/2021-March/000678.html

That will ensure the original source code is overseen by OpenJDK governance.

**Pull request merged**